### PR TITLE
Feature/checkstyle

### PR DESCRIPTION
**Pull request by vanderian**

### CODE REVIEW

File: library/src/main/java/ch/decent/sdk/utils/ElGamal.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ import java.math.BigInteger
 object ElGamal {
 
   private val modulus = BigInteger("11760620558671662461946567396662025495126946227619472274601251081547302009186313201119191293557856181195016058359990840577430081932807832465057884143546419")
-  private val generator = 3.toBigInteger()
+  @Suppress("MagicNumber") private val generator = 3.toBigInteger()
```
Comment (lukas1) [inline library/src/main/java/ch/decent/sdk/utils/ElGamal.kt]: This Suppress rule shows up quite a bit. Why not just set the lint check to ignore this rule?

File: library/src/main/java/ch/decent/sdk/api/ValidationApi.kt [hunk 2]
```diff
@@ -1,11 +1,24 @@
+@file:Suppress("TooManyFunctions", "LongParameterList")
```
Comment (lukas1) [inline library/src/main/java/ch/decent/sdk/api/ValidationApi.kt]: we repeat this rule a lot. How about setting up linter so that this is not a problem?

Comment (petervanderkadecent) [reply library/src/main/java/ch/decent/sdk/api/ValidationApi.kt]: Agree on this.

File: library/src/main/java/ch/decent/sdk/utils/ElGamal.kt (hunk 1 above)
Comment (vanderian) [reply library/src/main/java/ch/decent/sdk/utils/ElGamal.kt]: it is used 12 times, further refactor could lower it to half
magic numbers should be avoided, there are exclusions for
`    ignoreNumbers: '-1,0,1,2,8,10,0xFF' `
in some cases specific cases you want suppress this check
i do not see a reason to disable this check

File: library/src/main/java/ch/decent/sdk/api/ValidationApi.kt (hunk 2 above)
Comment (vanderian) [reply library/src/main/java/ch/decent/sdk/api/ValidationApi.kt]: these rules are ignored only in `api` package and `Utils` class, where we break these rules by intention, but they should definitely apply for the rest of the code

**Pull request merged**